contract/src/offerItems.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @file Contract to mint and sell a few Item NFTs at a time.
+ *
+ * We declare variables (including functions) before using them,
+ * so you may want to skip ahead and come back to some details.
+ * @see {start} for the main contract entrypoint
+ *
+ * @see {@link https://docs.agoric.com/guides/zoe/|Zoe Overview} for a walk-thru of this contract
+ * @see {@link https://docs.agoric.com/guides/js-programming/hardened-js.html|Hardened JavaScript}
+ * for background on `harden` and `assert`.
+ */
+// @ts-check
+
+import { Far } from '@endo/far';
+import { M, getCopyBagEntries, mustMatch } from '@endo/patterns';
+import { AmountMath, AssetKind } from '@agoric/ertp/src/amountMath.js';
+import { AmountShape } from '@agoric/ertp/src/typeGuards.js';
+import { atomicRearrange } from '@agoric/zoe/src/contractSupport/atomicTransfer.js';
+import '@agoric/zoe/exported.js';
+
+const { Fail, quote: q } = assert;
+
+// #region bag utilities
+/** @type { (xs: bigint[]) => bigint } */
+const sum = xs => xs.reduce((acc, x) => acc + x, 0n);
+
+/**
+ * @param {import('@endo/patterns').CopyBag} bag
+ * @returns {bigint[]}
+ */
+const bagCounts = bag => {
+  const entries = getCopyBagEntries(bag);
+  return entries.map(([_k, ct]) => ct);
+};
+// #endregion
```

Comment:
#region doesn't seem helpful in this context. I'd drop the comments or move the code to an import.

Reply:
feedback from @kbennett2000 was that #region is a well-known idiom from Visual Studio, so it was worthwhile. Do you two want to arm-wrestle and tell me how it comes out? :)

An import is what I might normally do, but that means the contract is no longer one file. That's OK if a walk-thru can assume readers grok `sum()` and `bagCount()` without looking at them. In fact, it could be a benefit to let folks know that they can split up their work.

Reply:
> Do you two want to arm-wrestle and tell me how it comes out? 

I will cede the ground. @kbennett2000 knows more than I do about how developers read code.